### ISSUE TITLE
Orders.GetOrders() is returning "Empty target resource argument found error" after updating library Version 1.7.2 > 1.7.18

### ISSUE THREAD

**Issue by smithburger** [abuzuhri/Amazon-SP-API-CSharp#743]
My application was using 1.7.2. I upgraded to 1.7.18 and Orders.GetOrders() stopped working. It is now returning "Empty target resource argument found error" no matter what inputs I give it. I have not had time to dive into this. If I get time and can figure it out I will post here. I wanted to say something now in case you or someone else knows what has changed recently. 

I was able to downgrade back to 1.7.2 for now. I tried to downgrade to 1.7.10 to see if I could figure out which update broke it. 1.7.10 threw the error as well. It is somewhere between .2 and .10.

```
AmazonConnection connection = new AmazonConnection(new AmazonCredential()
{
    AccessKey = ConfigurationManager.AppSettings["accessKey"],
    SecretKey = ConfigurationManager.AppSettings["secretKey"],
    RoleArn = ConfigurationManager.AppSettings["roleARN"],
    ClientId = ConfigurationManager.AppSettings["clientId"],
    ClientSecret = ConfigurationManager.AppSettings["clientSecret"],
    RefreshToken = ConfigurationManager.AppSettings["refreshToken"],
    MarketPlace = MarketPlace.US
});

var parameterOrderList = new ParameterOrderList
{
    CreatedAfter = DateTime.UtcNow.AddHours(-120),
    OrderStatuses = new List<OrderStatuses> { OrderStatuses.Unshipped },
    MarketplaceIds = new List<string> { MarketPlace.US.ID },
    IsNeedRestrictedDataToken = true
};

var amazonOrders = connection.Orders.GetOrders(parameterOrderList);
```

**Comment by michaelsowa** [comment]
I am also experiencing this issue.

I removed the AccessKey, SecretKey and RoleARN as they are no longer required.  However, when I set IsNeedRestrictedDataToken   to true I get that same message.

When IsNeedRestrictedDataToken    is false I do get orders data successfully.

**Comment by michaelsowa** [comment]
Just to confirm - I can accomplish this via Postman without an issue

**Comment by michaelsowa** [comment]
Further update - when I added this:

`RestrictedDataTokenRequest = new CreateRestrictedDataTokenRequest
            {
                restrictedResources = new List<RestrictedResource>
                {
                    new RestrictedResource
                    {
                        method = Method.GET.ToString(),
                        path = ApiUrls.OrdersApiUrls.Orders,
                        dataElements = new List<string> { "buyerInfo", "shippingAddress" }
                    }
                }
            }`

To my params, it works.

**Comment by shoecake** [comment]
> Further update - when I added this:
> 
> `RestrictedDataTokenRequest = new CreateRestrictedDataTokenRequest { restrictedResources = new List<RestrictedResource> { new RestrictedResource { method = Method.GET.ToString(), path = ApiUrls.OrdersApiUrls.Orders, dataElements = new List<string> { "buyerInfo", "shippingAddress" } } } }`
> 
> To my params, it works.

I have added this code and the getOrders now works, BUT no PII data is returned. I have tried to compare 1.7.2 and 1.7.18 and can't see the difference. Do you get PII data now? I do get it with 1.7.2 and on the getOrder call.

**Comment by shoecake** [comment]
Unfortunately this is not a complete solution. If the GetOrders call will return > 100 orders then the orders after the first 100 will not have the PII data in. I.e. the first call is fine, but if a NextToken is retuned then the subsequent requests have no PII data returned. I.e. the GetGetOrdersByNextTokenAsync function is not using the RestrictedDataToken

**Comment by mjw2022** [comment]
Was there a fix for this error, as I upgraded ver 1.7.27  today and have the same issue?  Thanks

**Comment by mjw2022** [comment]
> Was there a fix for this error, as I upgraded ver 1.7.27 today and have the same issue? Thanks

I was able overcome the issue by removing IsNeedRestrictedDataToken = true  BUT then get the address by using GetOrderAddress

  OrderAddress OrderAddress = amazonConnection.Orders.GetOrderAddress(order.AmazonOrderId);

**Comment by ouzhiman** [comment]
> Further update - when I added this:
> 
> `RestrictedDataTokenRequest = new CreateRestrictedDataTokenRequest { restrictedResources = new List<RestrictedResource> { new RestrictedResource { method = Method.GET.ToString(), path = ApiUrls.OrdersApiUrls.Orders, dataElements = new List<string> { "buyerInfo", "shippingAddress" } } } }`
> 
> To my params, it works.

Hello brother, have you been able to retrieve the address recently?

**Comment by ouzhiman** [comment]
> > Was there a fix for this error, as I upgraded ver 1.7.27 today and have the same issue? Thanks
> 
> I was able overcome the issue by removing IsNeedRestrictedDataToken = true BUT then get the address by using GetOrderAddress
> 
> OrderAddress OrderAddress = amazonConnection.Orders.GetOrderAddress(order.AmazonOrderId);

Hello brother, have you been able to retrieve the address recently?

**Comment by CarlosAdrianM** [comment]
> > > Was there a fix for this error, as I upgraded ver 1.7.27 today and have the same issue? Thanks
> > 
> > 
> > I was able overcome the issue by removing IsNeedRestrictedDataToken = true BUT then get the address by using GetOrderAddress
> > OrderAddress OrderAddress = amazonConnection.Orders.GetOrderAddress(order.AmazonOrderId);
> 
> Hello brother, have you been able to retrieve the address recently?

No, I think there is no fix or workaround for it. I was using version 1.7.9 until today, but now I have problems with credentials rotation in 1.7.9, so I am afraid I must look for a similar package without those problems.

**Comment by shoecake** [comment]
> The most similar to a workaround is:
> 
> ```
>             var conexion = AmazonApiOrdersService.ConexionAmazon();
>             var orders = await conexion.Orders.GetOrdersAsync(searchOrderList).ConfigureAwait(false);
>             foreach (var orderAdresss in orders)
>             {
>                 var address = await conexion.Orders.GetOrderAddressAsync(orderAdresss.AmazonOrderId).ConfigureAwait(false);
>                 orderAdresss.ShippingAddress = address.ShippingAddress;
>             }
>             return orders;
> ```

This is the best solution when there is > than 100 orders. For the first 100 orders you can manually add the Restricted Data Token:

RestrictedDataTokenRequest = new CreateRestrictedDataTokenRequest { restrictedResources = new List<RestrictedResource> { new RestrictedResource { method = Method.GET.ToString(), path = ApiUrls.OrdersApiUrls.Orders, dataElements = new List<string> { "buyerInfo", "shippingAddress" } } } }

It is the GetGetOrdersByNextTokenAsync function which does not use the RestrictedDataToken

**Comment by qaisarikram** [comment]
@abuzuhri can we please get a fix of this error?

**Comment by PoeticArcher** [comment]
Having the same issue. Buyers information isn't populating either when passing "buyerInfo" to the dataelements for restricted resource.
